go/ndt7-client/source/reader.go

```diff
@@ -0,0 +1,48 @@
+package source
+
+import (
+	"strings"
+	"time"
+
+	"github.com/apex/log"
+	"github.com/gorilla/websocket"
+)
+
+// logmeasurement logs a measurement received from the server. We trim the
```

Comment:
The same code appearing twice indicates the need for a measurementlogger interface, an instance of which you pass in to the Reader object. That way, test code can verify the the right stuff is read, and production code can format the message appropriately.

I also allows others who use this code a very convenient spot to put their own implementations.

Reply:
Right; I can create a specific package for this kind of code.